libopae/src/reconf.c

```diff
@@ -295,6 +366,13 @@ fpga_result __FPGA_API__ fpgaReconfigureSlot(fpga_handle fpga,
 		FPGA_ERR("Failed to clear port errors.");
 	}
 
+	if (!(flags & FPGA_RECONF_FORCE)) {
+		result = port_busy(fpga);
+		if (result != FPGA_OK) {
+			FPGA_ERR("Port device(s) may be in use.");
```

Comment:
If the Port is in use we should not allow reconfiguration and `goto out_unlock`
Also wouldn't it make more since to check if the Port is busy before trying to clear port_errors.

Reply:
* Yes, we should `goto out_unlock;`
* I was going to do this (check before clearing errors at first) but changed my mind for some reason.
  I can't think of why I changed my mind so I'll put it back in the order you recommend.